### PR TITLE
add check for block display name

### LINKED ISSUE #174: Check display name for all test blocks
## Description
Add check for same displayName in all test blocks. Throw exception if find same names

## Steps
- [ ] Add check
- [ ] Test it

### PR DESCRIPTION
**Pull request by alexander-kulikovskii**
update tests
closes #174

### CODE REVIEW

File: MapboxVision/src/test/java/com/mapbox/vision/dsl/VisionTestDsl.kt [hunk 1]
```diff
@@ -9,27 +9,53 @@ annotation class VisionDslMarker
...
+        addToSetOrThrowException(thenBlockDisplayName, thenBlockDisplayNames)
+
+        thenBlocks.add(DynamicTest.dynamicTest(thenBlockDisplayName, block))
+    }
+
+    private fun getBlockDisplayName(displayName: String, prefix: String) = "$prefix: $displayName"
+
+    private fun addToSetOrThrowException(displayName: String, displayNameSet: MutableSet<String>) {
+        if (displayName in displayNameSet) {
+            throw Exception("Block with name <$displayName> already exists")
```
Comment (yunikkk) [inline MapboxVision/src/test/java/com/mapbox/vision/dsl/VisionTestDsl.kt]: Shall we change it to 
`When|Then|Given with name ...` 
to ease understanding of exception?

Comment (alexander-kulikovskii) [reply MapboxVision/src/test/java/com/mapbox/vision/dsl/VisionTestDsl.kt]: Display name already contains such information.
E.x.:
Block with name <Given: TEST_NAME> already exists

File: MapboxVision/src/test/java/com/mapbox/vision/dsl/VisionTestDsl.kt [hunk 2]
```diff
@@ -9,27 +9,53 @@ annotation class VisionDslMarker
...
     operator fun invoke(block: TestContext.() -> Unit): List<DynamicContainer> {
         givenBlocks.clear()
+        givenBlockDisplayNames.clear()
         TestContext().run(block)
         return givenBlocks
     }
 
-    internal fun addGivenBlock(displayName: String) {
-        givenBlocks.add(DynamicContainer.dynamicContainer("Given: $displayName", whenBlocks.toMutableList()))
+    internal fun addGivenBlock(displayName: String, prefix: String = "Given") {
```
Comment (RingerJK) [inline MapboxVision/src/test/java/com/mapbox/vision/dsl/VisionTestDsl.kt]: why can we change `prefix`?  `addGivenBlock ` is obvious name for `Given`. And you could move `Given` to Consts or ENUM. The same for bottom functions 
```suggestion
    internal fun addGivenBlock(displayName: String) {
```

Comment (alexander-kulikovskii) [reply MapboxVision/src/test/java/com/mapbox/vision/dsl/VisionTestDsl.kt]: fix it

**Pull request merged**